src/webgpu/util/f32_interval.ts

```diff
@@ -197,47 +237,53 @@ export function ulpInterval(n: number, numULP: number): F32Interval {
 }
 
 /** For a point operation run its impl over the input that may be an interval */
-function runPointImpl(x: F32Interval, impl: PointToInterval): F32Interval {
+function runPointOp(x: F32Interval, op: PointToIntervalOp): F32Interval {
   if (x.isPoint()) {
-    return roundAndFlushPointToInterval(x.begin, impl);
+    return roundAndFlushPointToInterval(x.begin, op);
+  }
+
+  if (op.extrema !== undefined) {
+    x = op.extrema(x);
   }
   return F32Interval.span(
-    roundAndFlushPointToInterval(x.begin, impl),
-    roundAndFlushPointToInterval(x.end, impl)
+    roundAndFlushPointToInterval(x.begin, op),
+    roundAndFlushPointToInterval(x.end, op)
   );
 }
 
 /** For a binary operation run its impl over the inputs  that may be intervals */
-function runBinaryImpl(x: F32Interval, y: F32Interval, impl: BinaryToInterval): F32Interval {
+function runBinaryOp(x: F32Interval, y: F32Interval, op: BinaryToIntervalOp): F32Interval {
+  if (op.extrema !== undefined) {
+    [x, y] = op.extrema(x, y);
```

Comment:
I guess I'm confused.  For atan2, when the passed-in y interval contains 0, the resulting y interval is just [0,0].  
I'm not sure how that percolates through the rest of the calculation. Do we get the span over undefined values?

Reply:
{[0,0], x} will be passed into atan2, which will hit the special case for y === 0, which will create a span of {4096 * ulp(π), 4096 * ulp(0), 4096 * ulp(π)} (The zero term could be dropped since the other terms are on either side)

So the acceptance interval will be [-π, π].

Honestly atan2 is a hot mess of a trig function.

Taking a look at what Vulkan does. It looks like they are differentiating between the different quadrants based on the signs of y & x, and considering [0, 0] as undefined.

I think I need to rewrite this a bit given what Vulkan is doing and looking at the definition of atan2, I will put up another patch with corrected calculations that we can discuss in detail.

Reply:
I think I have this cleaned up now. I have ended up implementing:
y != 0 && x != 0 -> 4096 * ulp(Math.atan2(y, x))
y == 0 && x == 0 -> Infinite (undefined)
y == 0 && x != 0 -> [-π, π]

I think in theory we should be able to use x > 0 vs x < 0 to distinguish between positive/negative π cases, but everytime I try that either SwiftShader or Nvidia will start failing tests due to having the other sign for some x values.